client/src/js/components/bhReportSource.js

```diff
@@ -0,0 +1,48 @@
+angular.module('bhima.components')
+.component('bhReportSource', {
+  bindings : {
+    onSelectCallback : '&?',
+    label : '@?',
+    name : '@?',
+
+    // allows the controller to set a default value
+    value : '<'
+  },
+  template :
+    '<div class="form-group">' +
+      '<label class="control-label" translate>{{$ctrl.label}}</span></label>' +
+      '<select ng-change="$ctrl.onSelect($ctrl.value)" class="form-control" ng-model="$ctrl.value" ng-options="source.id as (source.label | translate) for source in $ctrl.sources" required>' +
+        '<option value="" disabled translate>FORM.SELECT.SOURCE</option>' +
+      '</select>' +
+      '{{$ctrl.invalue}}' +
+    '</div>',
+  controller : bhReportSourceController
+});
+
+bhReportSourceController.$inject = [];
+
+function bhReportSourceController() {
+  var ctrl = this;
+  var DEFAULT_SOURCE = 1;
+
+  ctrl.label = ctrl.label || 'FORM.SELECT.SOURCE';
```

Comment:
This initialization code should go in `$onInit()` to make the component clearer and more testable.  The idea is that `$onInit` fires after the bindings are initialized ... so you are guaranteed that this will default to `'FORM.SELECT.SOURCE'` _after_ the component has bound the `label` property and not before.

It also in the only way to construct components in Angular 2+.  [See this document for more information](https://angular.io/docs/ts/latest/guide/lifecycle-hooks.html#!#oninit)